src/libasr/pass/intrinsic_array_function_registry.h

```diff
@@ -1436,6 +1438,316 @@ namespace Cshift {
 
 } // namespace Cshift
 
+namespace Spread {
+    static inline void verify_args(const ASR::IntrinsicArrayFunction_t &x,
+            diag::Diagnostics &diagnostics) {
+        ASRUtils::require_impl(x.n_args == 3,
+            "`spread` intrinsic accepts 3 arguments",
+            x.base.base.loc, diagnostics);
+        ASRUtils::require_impl(x.m_args[0], "`source` argument of `spread` "
+            "cannot be nullptr", x.base.base.loc, diagnostics);
+        ASRUtils::require_impl(x.m_args[1], "`dim` argument of `spread` "
+            "cannot be nullptr", x.base.base.loc, diagnostics);
+        ASRUtils::require_impl(x.m_args[2], "`ncopies` argument of `spread` "
+            "cannot be nullptr", x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_Spread(Allocator &al, const Location &loc,
+            ASR::ttype_t *type, Vec<ASR::expr_t*> &args, diag::Diagnostics& diag) {
+        ASRBuilder b(al, loc);
+        int dim_val = ASR::down_cast<ASR::IntegerConstant_t>(ASRUtils::expr_value(args[1])) -> m_n;
+        if (all_args_evaluated(args) &&
+            (extract_n_dims_from_ttype(expr_type(args[0])) == 1) && (dim_val == 1 or dim_val == 2)) {
```

Comment:
This line gives the error message on Windows:
```
2024-07-02T06:52:30.6302614Z D:\a\lfortran\lfortran\lfortran-0.37.0-31-g4577497bc\src\libasr\..\libasr/pass/intrinsic_array_function_registry.h(1460): error C2146: syntax error: missing ')' before identifier 'or'
2024-07-02T06:52:30.6322973Z D:\a\lfortran\lfortran\lfortran-0.37.0-31-g4577497bc\src\libasr\..\libasr/pass/intrinsic_array_function_registry.h(1460): error C2143: syntax error: missing ';' before '{'
```
And to fix it, you can't use `or`:
```suggestion
            (extract_n_dims_from_ttype(expr_type(args[0])) == 1) && (dim_val == 1 || dim_val == 2)) {
```

Reply:
Oh I get this. Thank you!